packages/ckeditor5-engine/src/view/view.ts

```diff
@@ -776,3 +797,30 @@ export type ViewRenderEvent = {
 	name: 'render';
 	args: [];
 };
+
+/**
+ * An event fired at the moment of {@link module:engine/view/view~View#scrollToTheSelection} being called. It
+ * carries two kinds of data in its payload (`args`):
+ *
+ * * The first argument is the object containing data that will be passed down to the
+ *   {@link module:utils/dom/scroll~scrollViewportToShowTarget} helper. If some event listeners modifies it, it can adjust the behavior of
+ *   the scrolling (e.g. include additional `viewportOffset`).
+ * * The second argument corresponds to the original arguments passed to {@link module:utils/dom/scroll~scrollViewportToShowTarget}.
+ *   It allows listeners to re-execute the `scrollViewportToShowTarget()` method with its original arguments if there is such a need,
+ *   for instance, if the integration requires re–scrolling after certain interaction.
+ *
+ * @eventName ~View#scrollToTheSelection
+ */
+export type ViewScrollToTheSelectionEvent = {
+	name: 'scrollToTheSelection';
+	args: [
```

Comment:
It looks very ugly in API docs. I'd extract both arguments and define them as separate interfaces. One example: `DowncastReduceChangesEvent` and `DowncastReduceChangesEventData`.

Similarly, `scrollToTheSelection` parameter should be defined separately.